provider/resources.go

```diff
@@ -538,25 +538,22 @@ func arrayValue(vars resource.PropertyMap, prop resource.PropertyKey, envs []str
 	return vals
 }
 
-func durationFromConfig(vars resource.PropertyMap, prop resource.PropertyKey, envs []string) (time.Duration, error) {
+// returns a pointer so we can distinguish between a zero value and a missing value
```

Comment:
Nit: slight preference of adding more Go files in under ./provider instead of growing "resources.go" it's very big already, but certainly not blocking on this!